### PR TITLE
Sequence diagram examples

### PR DESCRIPTION
**Pull request by kmozsi**
Contains example sequence diagrams for _machine_, _monitoring_, _producer-consumer_ and _train_ demos. 
Unfortunately, the other demo models are currently not testable (see #590). 

Do not merge before #564, #579 and #584.

### CODE REVIEW

File: examples/demo/producer_consumer/src/producer_consumer/j/test/ProducerConsumerSequenceDiagramBase.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+public abstract class ProducerConsumerSequenceDiagramBase extends SequenceDiagram {
+
+	@Position(2)
+	Storage storage;
+
+	@Position(1)
+	Producer p1;
+
+	@Position(1)
```
[the remark this answers is not in the document]
Comment (kmozsi) [reply examples/demo/producer_consumer/src/producer_consumer/j/test/ProducerConsumerSequenceDiagramBase.java]: No, they do not need to be unique, as positions are handled as priorities. Lifelines with the same priority appear in the order of declaration.

File: examples/demo/machine/src/machine1/j/test/Machine1SequenceDiagram.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import hu.elte.txtuml.api.model.seqdiag.SequenceDiagram;
+import machine1.j.model.Machine;
+import machine1.j.model.User;
+import machine1.j.model.associations.Usage;
+import machine1.j.model.signals.ButtonPress;
+import machine1.j.model.signals.DoYourWork;
+
+public class Machine1SequenceDiagram extends SequenceDiagram {
+
+	Machine m;
```
[the remark this answers is not in the document]
Comment (kmozsi) [reply examples/demo/machine/src/machine1/j/test/Machine1SequenceDiagram.java]: It is not necessary to define positioning if the user does not want to care about lifeline ordering.

**Comment by djnemeth** [comment]
Jenkins, retest this, please.

**Comment by djnemeth** [comment]
Jenkins, retest this, please.

**Pull request merged**